src/cascadia/TerminalSettingsModel/PowershellInstallationProfileGenerator.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "PowershellInstallationProfileGenerator.h"
+#include "DynamicProfileUtils.h"
+
+#include <LibraryResources.h>
+
+static constexpr std::wstring_view POWERSHELL_ICON{ L"ms-appx:///ProfileIcons/pwsh.png" };
+static constexpr std::wstring_view POWERSHELL_ICON_64{ L"ms-appx:///ProfileIcons/Powershell_black_64.png" };
+
+namespace winrt::Microsoft::Terminal::Settings::Model
+{
+    std::wstring_view PowershellInstallationProfileGenerator::Namespace{ L"Windows.Terminal.InstallPowerShell" };
+
+    std::wstring_view PowershellInstallationProfileGenerator::GetNamespace() const noexcept
+    {
+        return Namespace;
+    }
+
+    std::wstring_view PowershellInstallationProfileGenerator::GetDisplayName() const noexcept
+    {
+        return RS_(L"PowerShellInstallationProfileGeneratorDisplayName");
+    }
+
+    std::wstring_view PowershellInstallationProfileGenerator::GetIcon() const noexcept
+    {
+        return POWERSHELL_ICON_64;
+    }
+
+    void PowershellInstallationProfileGenerator::GenerateProfiles(std::vector<winrt::com_ptr<implementation::Profile>>& profiles)
+    {
+        auto profile{ CreateDynamicProfile(RS_(L"PowerShellInstallationProfileName")) };
+        profile->Commandline(winrt::hstring{ fmt::format(FMT_COMPILE(L"cmd /k winget install --interactive --id Microsoft.PowerShell & echo. & echo {} & exit"), RS_(L"PowerShellInstallationInstallerGuidance")) });

```

Comment:
```suggestion
        profile->Commandline(winrt::hstring{ fmt::format(FMT_COMPILE(L"cmd /k winget install --interactive --id Microsoft.PowerShell --source winget & echo. & echo {} & exit"), RS_(L"PowerShellInstallationInstallerGuidance")) });
```

If this is done on a machine where WinGet hasn't been invoked before, the user will be prompted to accept `msstore` agreements first which don't seem relevant for this use case. Suggest to add explicit source argument so that user isn't prompted

Reply:
![image](https://github.com/user-attachments/assets/daefbcf4-5f27-453b-80f0-4d9af2473046)

Reply:
The Terminal shouldn't pass explicit agreements for the msstore source. That's something an IT administrator can do for their enterprise users, or an individual can do on their own, but applications shouldn't pass that on behalf of an end user. Just an FYI :)

Reply:
Yeah that's why I didn't suggest the `--accept-source-agreements` since we require agreements to be interactive

Reply:
When doing *any* programmatic winget operations, the source should *always* be provided, as well as explicitly specifying the Identifier, not just the identifier as a query string.